### ISSUE TITLE
macos feature request: swap back control, command keys

### ISSUE THREAD

**Issue by neeasade** [qutebrowser/qutebrowser#6149]
tl;dr I /think/ I'm looking for the inverse of https://github.com/qutebrowser/qutebrowser/issues/3697

I'm new to macos, where command seems to be treated like control in other places. My current approach is to use the control key as the command key, and then my muscle memory from elsewhere seems to "just work". This in turn means that the control key has become my "super" key, which I'm using for stuff like window management and misc workflow junk. qutebrowser seems to have this situation inverted though, so I can't for example use "command backspace" in the status bar to edit the url, because that's not the "control backspace" equivalent it appears to be in other places. 

Is there a way to treat command as control in QB? or should I do like a loop-remap thing in my own config.py

**Comment by The-Compiler** [comment]
To be honest I never really understood how that swapping in Qt worked exactly. What I know is that we do have to swap it back so that `Ctrl` in custom keybindings actually means the Ctrl key, and `Super` (which qutebrowser also aliases as `Cmd`) means Cmd.

In 1132324db9d665ba0f0143fb1d071995a3fc7715, I added `self.setAttribute(Qt.AA_MacDontSwapCtrlAndMeta, True)` in `app.py`, which is documented as:

> On macOS by default, Qt swaps the Control and Meta (Command) keys (i.e., whenever Control is pressed, Qt sends Meta, and whenever Meta is pressed Control is sent). When this attribute is true, Qt will not do the flip. QKeySequence::StandardKey will also flip accordingly (i.e., QKeySequence::Copy will be Command+C on the keyboard regardless of the value set, though what is output for QKeySequence::toString() will be different).

The Ctrl-Backspace binding is one added by Qt itself, rather than qutebrowser. Based on that description, I'd expect it to still act the same after setting that attribute - but maybe it's not a `QKeySequence::StandardKey` but hardcoded as `Ctrl-Backspace` somewhere in the `QLineEdit` implementation or so. I'd have to check what's going on exactly.

Before the change above, the swapping was done manually for the keybindings because of [QTBUG-51293](https://bugreports.qt.io/browse/QTBUG-51293) - perhaps it'd make sense to go back to that, to make sure any shortcuts added by Qt internally actually behave like you'd expect on macOS.

Could you try v1.14.1 which doesn't contain that change (ideally launching it manually with something like `qutebrowser.app/Contents/MacOS/qutebrowser --temp-basedir` - though that path is from memory)? I suspect it will behave in the way you describe.

**Comment by The-Compiler** [comment]
I dug into this some more, and it looks like Qt internally [does use QKeySequence](https://github.com/qt/qtbase/blob/v5.15.2/src/widgets/widgets/qwidgetlinecontrol.cpp#L1844-L1849), but then also handles [handles it manually](https://github.com/qt/qtbase/blob/v5.15.2/src/widgets/widgets/qwidgetlinecontrol.cpp#L1881-L1888)? Confusingly, QKeySequence [seems to have this defined](https://github.com/qt/qtbase/blob/v5.15.2/src/gui/kernel/qplatformtheme.cpp#L328-L329) as Alt-Backspace?

@neeasade You say:

> My current approach is to use the control key as the command key, and then my muscle memory from elsewhere seems to "just work". This in turn means that the control key has become my "super" key, which I'm using for stuff like window management and misc workflow junk.

Which confuses me - should the first sentence say "to use the command key as the control key"?

To clarify, what I understood is that:

- You use Mac's cmd (the key next to space) instead of ctrl, i.e. you would expect cmd-backspace to delete a word.
- You use Mac's ctrl (the left-most key) for window management
- However, in qutebrowser, cmd-backspace doesn't delete a word, only ctrl-backspace does.

Is that correct? Also, any chance you could give v1.14.1 a try as mentioned above?

**Comment by The-Compiler** [comment]
I've asked @RJ722 (who owns a Mac) to do some tests:

![image](https://user-images.githubusercontent.com/625793/108899194-71794e00-7618-11eb-97f9-dd84305f0f61.png)

So I suppose the fix for #6174 will indeed fix this as well.
